Update: Create Formula (#180)
ZupIT/ritchie-docs: content/en/Formulas/Create formulas.md

---
title: Create and update formulas
weight: 29
description: >-
  In this section, you will find how to create and update formulas on Ritchie and test them.
---

---

## **How to create?**

> ### **Run the formula creation command**

Reviewer: ```suggestion
### **Create your formula**
```

Run the following command to create a formula:

```text
rit create formula
```

{{% alert color="warning" %}}

> To run this command you need to have some template in your Ritchie's repos.

Reviewer 2: ```suggestion
To run this command, you need to have some templates in your Ritchie's repository.
```

> To run this command you need to have some template in your Ritchie's repos.

Reviewer: Missing comma after "command"

{{% /alert %}}

Then, you'll have to inform those inputs before running the formula:

1. The **command** (following the **pattern `rit + group + verb + noun`** to respect the [**tree pattern**]({{< ref path="Reference/Glossary" >}})).
2. The formula's **programming language**.
3. **The workspace** used to save formula's files.

{{% alert color="info" %}}

> When you finished creating your formula, it will will be built according to the template and sent to a specific repository according to the selected **workspace**.

Reviewer 2: ```suggestion
When you have finished creating your formula, it will be built according to the template and sent to a specific repository according to the selected **workspace**.
```

> When you finished creating your formula, it will will be built according to the template and sent to a specific repository according to the selected **workspace**.

Reviewer: When you **have** finished creating your formula, it will ~will~ be built according to the template and sent to a specific repository according to the selected **workspace**.

]
```

{{% /alert %}}

Example: `rit demo create formula`

![](/shared/rit_create_formula.gif)

### **Update your formula**

You can update the formula by modifying their workspace files.

> To more informations about which files to modify go to [**organize the formula folder**]({{< ref path="formulas/Organize the formula folder" >}}).

Reviewer: To more *information* about which files to modify, go to...

> To more informations about which files to modify go to [**organize the formula folder**]({{< ref path="formulas/Organize the formula folder" >}}).

Reviewer 2: ```suggestion
For more information about which files to modify, go to [**organize the formula folder**]({{< ref path="formulas/Organize the formula folder" >}}).
```

When you run the formula for the first time after modifying the files in the workspace, Ritchie will apply the modification and rebuild the formula automatically.

{{% alert color="warning" %}}

Don't forget to update the unit tests.

{{% /alert %}}

### **Test your formula**

You can test the formula directly after its creation. That's possible because the **`rit create formula`** command also builds the formula automatically (it generates the formula's executable files and add them on **.rit folder**).

In that case, the formula **"Hello World" template** will be executed as you can see below:

![](/shared/rit_demo_hello-world.gif)

> This template is composed of **4 inputs parameters**, that represents the currently available kinds of inputs to use Ritchie:

Reviewer: A suggestion to remove passive voice:

They exist **4 inputs parameters** that's compose this template, that represents the currently available kinds of inputs to use Ritchie

> This template is composed of **4 inputs parameters**, that represents the currently available kinds of inputs to use Ritchie:

Reviewer 2: ```suggestion
This template has 4 inputs parameters, that represent the currently available kinds of inputs to use Ritchie.
```

Reviewer 2: @HenriqueZup another suggestion to avoid passive voice :)
